gnucash/gnome/gnc-plugin-page-register.cpp

```diff
@@ -4702,12 +4704,23 @@ gnc_plugin_page_register_cmd_goto_date (GSimpleAction *simple,
 
     for (GList *lp = splits; lp; lp = lp->next)
     {
-        if (xaccTransGetDate (xaccSplitGetParent (GNC_SPLIT(lp->data))) >= date.value())
+        Split *split = GNC_SPLIT(lp->data);
+        Transaction *trans = xaccSplitGetParent (split);
+
+        // if gl register, there could be blank splits from other open registers
+        // included in split list so check for and ignore them
+        if (xaccTransCountSplits (trans) != 1 && xaccSplitGetAccount (split) != nullptr)
         {
-            gnc_split_reg_jump_to_split (gsr, GNC_SPLIT(lp->data));
-            break;
+            if (xaccTransGetDate (trans) >= date.value())
+            {
+                found = true;
+                gnc_split_reg_jump_to_split (gsr, split);
+                break;
+            }
         }
     }
+    if (!found)
+       gnc_split_reg_jump_to_blank (gsr);
```

Comment:
haven't tested:

```patch
diff --git a/gnucash/gnome/gnc-plugin-page-register.cpp b/gnucash/gnome/gnc-plugin-page-register.cpp
index 92732a4be3..ff7a1f7d1c 100644
--- a/gnucash/gnome/gnc-plugin-page-register.cpp
+++ b/gnucash/gnome/gnc-plugin-page-register.cpp
@@ -4674,6 +4674,15 @@ gnc_plugin_page_register_cmd_blank_transaction (GSimpleAction *simple,
     LEAVE (" ");
 }
 
+static bool
+find_after_date (Split *split, gpointer user_data)
+{
+    auto trans = xaccSplitGetParent (split);
+    return (xaccTransCountSplits (trans) != 1 &&
+            xaccSplitGetAccount (split) != nullptr &&
+            xaccTransGetDate (trans) >= GPOINTER_TO_INT(user_data));
+}
+
 static void
 gnc_plugin_page_register_cmd_goto_date (GSimpleAction *simple,
                                         GVariant      *paramter,
@@ -4700,14 +4709,12 @@ gnc_plugin_page_register_cmd_goto_date (GSimpleAction *simple,
     splits = g_list_copy (qof_query_run (query));
     splits = g_list_sort (splits, (GCompareFunc)xaccSplitOrder);
 
-    for (GList *lp = splits; lp; lp = lp->next)
-    {
-        if (xaccTransGetDate (xaccSplitGetParent (GNC_SPLIT(lp->data))) >= date.value())
-        {
-            gnc_split_reg_jump_to_split (gsr, GNC_SPLIT(lp->data));
-            break;
-        }
-    }
+    auto it = g_list_find_custom (splits, GINT_TO_POINTER (date.value()),
+                                  (GCompareFunc)find_after_date);
+    if (it)
+        gnc_split_reg_jump_to_split (gsr, GNC_SPLIT(it->data));
+    else
+        gnc_split_reg_jump_to_blank (gsr);
 
     g_list_free (splits);
     LEAVE (" ");
```

Reply:
ok patch amended, should be fine on 32-bit

Reply:
@christopherlam TBC the issue with GINT_TO/FROM_POINTER is that it's a 32-bit operation because int is a 32-bit type, at least on the architectures we're interested in. You can't use it with time64 even where pointers are 64-bit because you can't be sure that the compiler won't cast the time64 to int, discarding the top half, before casting the result to a pointer.

Reply:
> @christopherlam TBC the issue with GINT_TO/FROM_POINTER is that it's a 32-bit operation because int is a 32-bit type, at least on the architectures we're interested in. You can't use it with time64 even where pointers are 64-bit because you can't be sure that the compiler won't cast the time64 to int, discarding the top half, before casting the result to a pointer.

Is the amended patch any better? No GINT_ macros

Reply:
Yes, much. I just think it's important for you to understand the underlying problem with casting between ints and pointers. The same problem applies to casting between numeric types of different sizes like doubles to ints: Bits get lost.

Reply:
I have updated the PR with Chris's suggested change.